go/kbcmf/encrypt.go

```diff
@@ -160,10 +217,16 @@ func (es *encryptStream) init(sender BoxSecretKey, receivers [][]BoxPublicKey) e
 				return err
 			}
 
-			rkc := receiverKeysCiphertext{
-				KID:  kid,
-				Keys: ptec,
+			rkc := receiverKeysCiphertexts{
+				Keys:   ptec,
+				Sender: ske,
 			}
+
+			// Don't specify the receivers if this public key wants to hide
+			if !receiver.HideIdentity() {
+				rkc.KID = receiver.ToKID()
```

Comment:
I think we're OK if we're all visible or all hidden.  If the caller wants to do something else, there's rope to hang themselves with, i.e., to leak some information.  It will all be out in the open either way.  But all receiver groups are MAC'ed now, that also might eliminate some of this concern.